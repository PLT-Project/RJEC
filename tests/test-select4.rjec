func foo(a chan int, b chan int, c chan int) {
    val := 1313;
    for {
        select {
        case a <- 777:
            printi(777);
        case b <- 888:
            printi(888);
        case c <- val:
            printi(val);
        }
    }
}

func main() {
    a := make(chan int);
    b := make(chan int);
    c := make(chan int);
    yeet foo(a, b, c);
    printi(<-a);
    printi(<-b);
    printi(<-c);
}