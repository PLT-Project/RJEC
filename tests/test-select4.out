777
777
888
888
1313
1313